cosmos/providers/dbt/core/utils/adapted_subprocesshook.py

```diff
@@ -0,0 +1,108 @@
+# This hook has been refined from the Airflow SubprocessHook, offering an added functionality to the original.
+# It presents an alternative option to return the complete command output, as opposed to solely the last line from
+# stdout or stderr. This option proves to be highly beneficial for any text analysis that depends on the stdout or
+# stderr output of a dbt command.
+from __future__ import annotations
+
+import contextlib
+import os
+import signal
+from collections import namedtuple
+from subprocess import PIPE, STDOUT, Popen
+from tempfile import TemporaryDirectory, gettempdir
+
+from airflow.hooks.base import BaseHook
+
+SubprocessResult = namedtuple(
+    "SubprocessResult", ["exit_code", "output", "full_output"]
+)
```

Comment:
only other change I'd make is to rename this as well - and then in the local operator, make sure we use this type as the return value/parameter types in the local operator

Reply:
good catch ! changes done as suggested !